packages/rum-react/src/get-with-transaction.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+/*
+    Usage:
+        - Pure function: `withTransaction('name','route-change')(Component)`
+        - As a decorator: `@withTransaction('name','route-change')`
+ */
+
+import React from 'react'
+import hoistStatics from 'hoist-non-react-statics'
+
+function getWithTransaction(apm) {
+  if (!apm) {
```

Comment:
this feels unnecessary, We should just be exposing `withTransaction`.

Reply:
I think `if (!apm) {` is a mistake I will remove that. But the function `getWithTransaction`  is there so that we can pass an apm instance to it. The reason is that we want to make sure that each tests runs with fresh new apm instance.

Reply:
We should have `getWithTransaction` function as part of tests then, Having something in `src` to ease testing is not ideal.

Reply:
I'm not sure how can we pass the apm instance if we don't define it in the source. Would you mind elaborating?

Reply:
Something like this. 

```js
import { withTransaction } from "../"
import { createApmBase } from "@elastic/apm-rum"

let apmInstance
beforeEach(() => apmInstance = new createApmBase())

```

Reply:
That doesn't work, we need to pass the apm instance to withTransaction so that it uses that one.